libs/stocks/feature-shell/src/lib/stocks/stocks.component.ts

```diff
@@ -28,16 +33,36 @@ export class StocksComponent implements OnInit {
   constructor(private fb: FormBuilder, private priceQuery: PriceQueryFacade) {
     this.stockPickerForm = fb.group({
       symbol: [null, Validators.required],
-      period: [null, Validators.required]
-    });
+      period: [null, Validators.required],
+      dateFrom: [null, Validators.required],
+      dateTo: [null, Validators.required]
+    }, {validator: this.dateLessThan('dateFrom', 'dateTo')});
+  }
+
+  /**
+   * We feed from and to date and compare them
+   * then returns validation message
+   */
+  dateLessThan(from: string, to: string) {
+    return (group: FormGroup): {[key: string]: any} => {
+      const f = group.controls[from];
+      const t = group.controls[to];
+      if (f.value > t.value) {
+          return {
+          dates: "Date from should be less than Date to"
+        };
+      }
+      return {};
+    }
```

Comment:
This is custom date validator. please read comments from line 43 to 44 for its purpose